activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -78,6 +78,32 @@ def #{method_name}=(value)           # def includes_values=(value)
 
     alias extensions extending_values
 
+    # Allows preloading of associations on a relation or array of records,
+    # in the same way that #preload does. Mutates the +records+ argument.
+    # For example:
+    #
+    #   users = User.all.to_a
+    #   User.load_associations(users, :posts)
+    #   # SELECT "posts".* FROM "posts" WHERE "posts"."user_id" IN (1, 2, 3)
+    #
+    #   users.each do |user|
+    #     user.posts
+    #   end
+    def load_associations(records, *associations)
+      if records.is_a?(ActiveRecord::Relation)
+        records.preload(associations.flatten)
+      else
+        check_if_array_contains_valid_elements!(records)
+        check_if_method_has_arguments!(:load_associations, associations)
+
+        preloader = build_preloader
+        associations.each do |association|
+          preloader.preload records, association
+        end
+        records
+      end
+    end
```

Comment:
Still think a `load_association` (singular) alias would be nice for the single association case.